include/fpu.h

```diff
@@ -45,105 +45,105 @@ void FPU_ESC6_EA(Bitu func,PhysPt ea);
-        uint32_t lower;
-        int32_t upper;
-    } l;
+		uint32_t lower;
+		int32_t upper;
 #else
-    struct {
-        int32_t upper;
-        uint32_t lower;
-    } l;
+		int32_t upper;
+		uint32_t lower;
 #endif
-    int64_t ll;
-} FPU_Reg;
+	} l;
+	int64_t ll;
+};
 
-typedef struct {
-    uint32_t m1;
-    uint32_t m2;
-    uint16_t m3;
+struct FPU_P_Reg {
+	uint32_t m1 = 0;
+	uint32_t m2 = 0;
+	uint16_t m3 = 0;
 
-    uint16_t d1;
-    uint32_t d2;
-} FPU_P_Reg;
+	uint16_t d1 = 0;
+	uint32_t d2 = 0;
+};
 
-enum FPU_Tag {
+enum FPU_Tag : uint8_t {
 	TAG_Valid = 0,
 	TAG_Zero  = 1,
 	TAG_Weird = 2,
 	TAG_Empty = 3
 };
 
-enum FPU_Round {
-	ROUND_Nearest = 0,		
+enum FPU_Round : uint8_t {
+	ROUND_Nearest = 0,
 	ROUND_Down    = 1,
-	ROUND_Up      = 2,	
+	ROUND_Up      = 2,
 	ROUND_Chop    = 3
 };
 
-typedef struct FPU_rec {
-	FPU_Reg		regs[9];
-	FPU_P_Reg	p_regs[9];
-	FPU_Tag		tags[9];
-	uint16_t		cw,cw_mask_all;
-	uint16_t		sw;
-	uint32_t		top;
-	FPU_Round	round;
-} FPU_rec;
-
-#define L2E		1.4426950408889634
-#define L2T		3.3219280948873623
-#define LN2		0.69314718055994531
-#define LG2		0.3010299956639812
+struct FPU_rec {
+	FPU_Reg regs[9]      = {};
+	FPU_P_Reg p_regs[9]  = {};
+	FPU_Tag tags[9]      = {};
+	uint16_t cw          = 0;
+	uint16_t cw_mask_all = 0;
+	uint16_t sw          = 0;
+	uint32_t top         = 0;
+	FPU_Round round      = {};
+};
 
+constexpr auto L2E = 1.4426950408889634;
+constexpr auto L2T = 3.3219280948873623;
+constexpr auto LN2 = 0.69314718055994531;
+constexpr auto LG2 = 0.3010299956639812;
 
 extern FPU_rec fpu;
 
 #define TOP fpu.top
 #define STV(i)  ( (fpu.top+ (i) ) & 7 )
 
-
-uint16_t FPU_GetTag(void);
+uint16_t FPU_GetTag();
 void FPU_FLDCW(PhysPt addr);
 
-static inline void FPU_SetTag(uint16_t tag){
-	for(Bitu i=0;i<8;i++)
-		fpu.tags[i] = static_cast<FPU_Tag>((tag >>(2*i))&3);
+static inline void FPU_SetTag(const uint16_t tag)
+{
+	for (uint8_t i = 0; i < 8; ++i) {
+		fpu.tags[i] = static_cast<FPU_Tag>((tag >> (2 * i)) & 3);
+	}
 }
 
 static inline uint16_t FPU_GetCW()
 {
 	return fpu.cw;
 }
 
-static inline void FPU_SetCW(Bitu word)
+static inline void FPU_SetCW(const uint16_t word)
 {
-	fpu.cw = (uint16_t)word;
-	fpu.cw_mask_all = (uint16_t)(word | 0x3f);
-	fpu.round = (FPU_Round)((word >> 10) & 3);
+	fpu.cw          = word;
+	fpu.cw_mask_all = word | 0x3f;
+	fpu.round       = static_cast<FPU_Round>((word >> 10) & 3);
 }
 
 static inline uint16_t FPU_GetSW()
 {
 	return fpu.sw;
 }
 
-static inline void FPU_SetSW(Bitu word)
+static inline void FPU_SetSW(const uint16_t word)
 {
-	fpu.sw = (uint16_t)word;
+	fpu.sw = word;
 }
 
-static inline Bitu FPU_GET_TOP(void) {
-	return (fpu.sw & 0x3800)>>11;
+static inline uint8_t FPU_GET_TOP()
+{
+	return static_cast<uint8_t>((fpu.sw & 0x3800) >> 11);
 }
 
-static inline void FPU_SET_TOP(Bitu val){
+static inline void FPU_SET_TOP(const uint32_t val)
+{
 	fpu.sw &= ~0x3800;
-	fpu.sw |= (val&7)<<11;
+	fpu.sw |= static_cast<uint16_t>((val & 7) << 11);
```

Comment:
I don't think this helps with clarity. Why not extract `0x3800` into a const instead and reference in both the setter and getter?